src/engines/TrainElectricEngine.cpp

```diff
@@ -24,42 +23,352 @@ namespace godot {
+        ClassDB::bind_method(
+                D_METHOD("get_power_cable_steam_pressure"), &TrainElectricEngine::get_power_cable_steam_pressure);
+        ADD_PROPERTY(
+                PropertyInfo(Variant::FLOAT, "power/power_cable/steam_pressure"), "set_power_cable_steam_pressure",
+                "get_power_cable_steam_pressure");
     }
 
     void TrainElectricEngine::_do_fetch_state_from_mover(TMoverParameters *mover, Dictionary &state) {
         TrainEngine::_do_fetch_state_from_mover(mover, state);
 
         state["compressor_enabled"] = mover->CompressorFlag;
         state["compressor_allowed"] = mover->CompressorAllow;
-
         state["converter_enabled"] = mover->ConverterFlag;
         state["converted_allowed"] = mover->ConverterAllow;
         state["converter_time_to_start"] = mover->ConverterStartDelayTimer;
+        state["power_source"] = static_cast<int>(mover->EnginePowerSource.SourceType);
+        state["accumulator/recharge_source"] = static_cast<int>(mover->EnginePowerSource.RAccumulator.RechargeSource);
+        state["current_collector/max_voltage"] = mover->EnginePowerSource.MaxVoltage;
+        state["current_collector/max_current"] = mover->EnginePowerSource.MaxCurrent;
+        state["current_collector/max_collector_lifting"] = mover->EnginePowerSource.CollectorParameters.MaxH;
+        state["current_collector/min_collector_lifting"] = mover->EnginePowerSource.CollectorParameters.MinH;
+        state["current_collector/csw"] = mover->EnginePowerSource.CollectorParameters.CSW;
+        state["current_collector/min_main_switch_voltage"] = mover->EnginePowerSource.CollectorParameters.MinV;
+        state["current_collector/min_pantograph_tank_pressure"] = mover->EnginePowerSource.CollectorParameters.MinPress;
+        state["current_collector/max_pantograph_tank_pressure"] = mover->EnginePowerSource.CollectorParameters.MaxPress;
+        state["current_collector/overvoltage_relay"] = mover->EnginePowerSource.CollectorParameters.OVP;
+        state["current_collector/required_main_switch_voltage"] = mover->EnginePowerSource.CollectorParameters.InsetV;
+        state["transducer/input_voltage"] = mover->EnginePowerSource.Transducer.InputVoltage;
+        state["power_cable/source"] = static_cast<int>(mover->EnginePowerSource.RPowerCable.PowerTrans);
+        state["power_cable/steam_pressure"] = mover->EnginePowerSource.RPowerCable.SteamPressure;
     }
 
     void TrainElectricEngine::_do_update_internal_mover(TMoverParameters *mover) {
         TrainEngine::_do_update_internal_mover(mover);
 
-        // FIXME: test data
-        mover->EnginePowerSource.SourceType = TPowerSource::CurrentCollector;
-
+        mover->EnginePowerSource.SourceType = static_cast<TPowerSource>(power_source);
         mover->CompressorSwitch(compressor_switch_pressed);
         mover->ConverterSwitch(converter_switch_pressed);
+        switch (static_cast<TPowerSource>(power_source)) {
```

Comment:
hm. to zadziała tylko i wyłącznie wtedy, gdy PROPERTY_HINT z godota będzie zgadzał się z enumem TPowerSource. Wystarczy jedna różnica i kaplica. Tu powinien być switch na typ `power_soruce`. Uczynić go wartałoby enumem naszym, i jego użyć w switch-case. Z kolei w każdym case ustawiać moverowi enum wg TPowerSource.